### PR TITLE
Fix #2952: Add support for disabling practice tab

### LINKED ISSUE #2952: Need support for disabling the practice tab
Since there are very few questions available in production, we need to be able to disable the practice tab so that users don't run into a broken or confusing experience. This will be used in the next alpha release.

### PR DESCRIPTION
**Pull request by BenHenning**
## Explanation
Fix #2952

Introduces support for disabling the practice tab in the topic fragment. This is done via a new Dagger boolean qualified by ``EnablePracticeTab``. In order to easily disable the tab, ``TopicTab`` has been updated to actually compute/manage both tab positions and count based on whether the practice tab should be enabled.

Some other minor refactoring was done in ``TopicTab`` to try and consolidate tab-specific resources & strings into a single location which avoids extra ``when`` statements elsewhere.

I verified that TopicFragmentTest is also passing locally when run with Espresso on a Nexus 5X.

### CODE REVIEW

**Comment by rt4914** [comment]
@BenHenning Does this PR still needs a review?

**Comment by anandwana001** [comment]
I will review this today.

**Review by anandwana001 — APPROVED** [review]
LGTM, Thanks @BenHenning

**Comment by anandwana001** [comment]
https://github.com/oppia/oppia-android/pull/3072 blocked on this PR as #2986 has a lot of change around `topicTab` enum and TopicInfo which also requires here in this PR. 
@rt4914 Could you review this pr so we can merge it asap.

File: app/src/main/java/org/oppia/android/app/topic/revisioncard/PracticeTabModule.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.oppia.android.app.topic
+
+import dagger.Module
+import dagger.Provides
+
+/** Provides dependencies corresponding to the practice tab. */
+@Module
+class PracticeTabModule {
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/topic/revisioncard/PracticeTabModule.kt]: I think by mistake this file shifted to `revisioncard` package.

**Review by rt4914 — APPROVED** [review]
LGTM, one nit change and rest is blocked on tests.

**Comment by BenHenning** [comment]
Ack thanks for the notice of blocking @anandwana001. I'm working to get all upstream work this depends on in. Nearly everything before this PR is now in review (only #2985 isn't yet).

File: app/src/main/java/org/oppia/android/app/topic/revisioncard/PracticeTabModule.kt (hunk 1 above)
Comment (BenHenning) [reply app/src/main/java/org/oppia/android/app/topic/revisioncard/PracticeTabModule.kt]: Nice catch. Fixed (will be visible in the next push to this branch).

**Comment by BenHenning** [comment]
Thanks @anandwana001 & @rt4914 for the reviews.

@rt4914 given some new modules were migrated, your earlier comment, and the new test to verify the tab properly hides, PTAL again when you get a chance.

File: app/src/sharedTest/java/org/oppia/android/app/topic/TopicFragmentTest.kt [hunk 2]
```diff
@@ -214,8 +225,39 @@ class TopicFragmentTest {
...
+              R.id.topic_tabs_container
+            )
+          )
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testTopicFragment_practiceTabDisabled_practiceTopicTabIsNotDisplayedInTabLayout() {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/topic/TopicFragmentTest.kt:247]: I think there should be tests to check that on configuration-change the tab remains hidden/visible.

**Review by rt4914 — APPROVED** [review]
@BenHenning Overall LGTM. Suggested to add one test case. 
Not blocking this PR on that test suggestion.

**Comment by BenHenning** [comment]
Only Bazel failure was RevisionCardActivityLocalTest which seems to be due to a network flake (a Maven dependency failed to download). I think we can safely ignore this and avoid needing to re-run Bazel tests before merging this PR in.

File: app/src/sharedTest/java/org/oppia/android/app/topic/TopicFragmentTest.kt (hunk 2 above)
Comment (BenHenning) [reply app/src/sharedTest/java/org/oppia/android/app/topic/TopicFragmentTest.kt:247]: Nice idea. Done.

**Comment by BenHenning** [comment]
Thanks for the review! I added the suggest test change & verified that TopicFragmentTest is passing locally with Espresso.

Note I'll probably cancel the Bazel tests per my comment above & verifying tests passing locally. I'd prefer to get the whole chain in quickly since I'll be verifying that Bazel is passing at the end (after #3116 is merged).

**Pull request merged**